### ISSUE TITLE
docker container / job status mess up (vagrant getting started test setup)

### ISSUE THREAD

**Issue by far-blue** [hashicorp/nomad#801]
Having followed the vagrant-based intro tutorial (successfully) and having 2 agents and the server running on one vm with a shared docker-engine I wanted to understand what might happen if a docker container died or was killed outside of the nomad environment. So I directly killed one of the 3 example redis containers via docker. I expected nomad to re-schedule the container or at least report that it had failed. However, `nomad status` still reported the container as active and running.

Is this expected behaviour?

**Comment by diptanu** [comment]
Yeah if the job was ran by the service scheduler, nomad would have just restarted your redis container when it died but it would still report that the allocation is running. Nomad would move the state of the allocation to dead only when it can never restart that allocation.

**Comment by far-blue** [comment]
I'm not sure I understand you reply. I was using the job configuration generated by `nomad init` (as per the tutorial) which, as I understand it, uses the default job type of `service`. Nomad didn't reschedule it and also didn't report it as dead. This left nomad (incorrectly) thinking the job was fine but docker (correctly) reporting that only 2 of the three containers were running.

I also tried removing the container with `docker rm` in case simply stopping wasn't enough for nomad to notice. Nomad still didn't reschedule the container and continued to report it as running.

**Comment by diptanu** [comment]
So Nomad is supposed to restart the redis container when you kill it. If it didn't do it, then it's a bug. Can you please paste the logs of the nomad client where you are killing the container and nomad isn't restarting? This will help us troubleshoot the issue.

**Comment by far-blue** [comment]
I've repeated the sequence and in the process of creating the debug info for you I noticed that docker container is restarted, eventually. The delay is around 30 seconds which leads me to assume the `delay` configuration value in the job is controlling this delay. So it does look like nomad is working as expected :)

My initial understanding of the `delay` config value was that it only applied when a start or a restart failed and a second or subsequent attempt to start the container was made within the `interval` window (to rate-limit repeated start/fail cycles if there was a problem). I now believe it applies in all cases - even when a container has been running for longer than the `interval` window - so any failure, no matter the cause, results in a `delay` wait before it is restarted.

So I guess if I want my container to restart quickly if it fails (or, in future, if a server / node fails) then I need to set the `delay` value low (e.g. 2s) -  with the understanding this would also result in a rapid cycle of failed restarts if the container was to fail on restart.
